test/grep.js

```diff
@@ -162,6 +180,30 @@ test('-i option', t => {
   t.is(result.split('\n').length - 1, 3);
 });
 
+test('one directory provided with -r option', t => {
+  const result = shell.grep('-r', 'test', 'test/resources');
+  t.falsy(shell.error());
+  t.is(result.split('\n').length, 29);
+});
+
+test('multiple directories or files provided', t => {
+  const result = shell.grep('-r', 'test', 'test/resources', 'test/resources/a.txt', 'test/resources/file2.txt', 'test/utils/');
+  t.falsy(shell.error());
+  t.is(result.split('\n').length, 39);
+});
+
+test('directory provided in glob form', t => {
+  const result = shell.grep('-r', 'test', 'test/r*');
+  t.falsy(shell.error());
+  t.is(result.split('\n').length, 66);
+});
+
+test('works with array syntax', t => {
+  const result = shell.grep('-r', 'test', ['test/r*', 'package.json'], 'scripts');
```

Comment:
Please do not include anything outside of the resources/ folder in the test (package.json, scripts, etc.). It makes the test much more difficult to maintain if we do unrelated changes in package.json etc.

I think the same goes for `test/r*` because that matches `test/rm.js`. Instead of this, how about adding something like `test/resources/grep2` and testing the glob `test/resources/g*` if you want to test globbing?